src/complete.cpp

```diff
@@ -226,7 +226,7 @@ completion_t &completion_t::operator=(const completion_t &him) {
 
 bool completion_t::is_naturally_less_than(const completion_t &a, const completion_t &b) {
     //for this to work, stable_sort must be used because results aren't interchangeable
-    if ((a.flags & COMPLETE_DONT_SORT) != 0 && (b.flags & COMPLETE_DONT_SORT) != 0) {
+    if (a.flags & b.flags & COMPLETE_DONT_SORT) {
```

Comment:
Nifty but subtle enough that it probably requires a comment. How often have you seen a sequence of bitwise "and" operators like this? Despite programming in C for 30+ years it took me a few seconds to realize this was equivalent to your original formulation. At first glance I thought this would be true if either argument had the flag set. 🙁  On the other hand had I seen this statement without the context of what led up to it I might not have jumped to the wrong conclusion. So, your call as to whether a comment is warranted.